src/vite/island-components.ts

```diff
@@ -29,27 +29,34 @@ import {
   memberExpression,
   importDeclaration,
   importSpecifier,
+  exportNamedDeclaration,
+  exportSpecifier,
 } from '@babel/types'
 import { parse as parseJsonc } from 'jsonc-parser'
 // eslint-disable-next-line node/no-extraneous-import
 import type { Plugin } from 'vite'
 
-function addSSRCheck(funcName: string, componentName: string) {
+function isComponentName(name: string) {
+  return /^[A-Z]/.test(name) && /[a-z]/.test(name)
```

Comment:
I think the component name is `/^[A-Z][A-Za-z0-9]+/`. What do you think?

Reply:
I didn't explain it well enough.

I wanted to exclude capital-only exports from the island component, didn't I? What do you think?

```
export const LIMIT = 10;.
```

Reply:
Well, but it is certainly better not to include `_`. I prefer something like this.

```ts
/^[A-Z][A-Za-z0-9]*[a-z][A-Za-z0-9]*$/
```

I will change it to this later. Wait a minute.

Reply:
How about 130c642?

Reply:
@usualoma 

> How about 130c642?

Good! I'd like to go with it.

How about writing the test? I want to have a test, though it's simple logic. But, I don't have any good idea where we should put it. Exporting `isComponentName` and writing test in `src/vite/island-components.test.ts`? Or creating `src/vite/utils.ts` and putting the logic into it? Do you have any thoughts?

Reply:
Thank you. You're right, I should add a test.

The test for this change already existed and there was no test for `Utility_Fn`, so I added e8789be.
How about this?

https://github.com/usualoma/honox/blob/e8789be86d993242907fe0eaf697b0db0d3fc40a/src/vite/island-components.test.ts#L77-L88
https://github.com/usualoma/honox/blob/e8789be86d993242907fe0eaf697b0db0d3fc40a/src/vite/island-components.test.ts#L171-L186

Reply:
Thanks. But honestly, I want a unit test for `isComponentName`.

Reply:
Well, I can see why you want to test it on the list, I'll add a few things, just wait and see.

Reply:
8365f1b added!

I think what you are trying to say is that "whether to treat it as a component or not is a very important external specification, so it should be an independent test, both to confirm the implementation and to explain it to the user".

For me, 8365f1b seems sufficient, but if I were to go further, I would suggest that it should be independent, as you say, as follows.

> Or creating `src/vite/utils.ts` and putting the logic into it

Opinion me, please.

Reply:
Oops, fix typo 35aa726

Reply:
> whether to treat it as a component or not is a very important external specification, so it should be an independent test, both to confirm the implementation and to explain it to the user

Yes! Exactly. Thank you for saying what I wanted to say!

https://github.com/honojs/honox/commit/8365f1b070e25abe7ae0855ca3c3cfc8413cc630 is good for me. This covers all patterns and shows the specification. Let's go with it. Sorry for taking your time.

Reply:
I wish I had been aware of this from the beginning, but I am sorry for taking up so much of your time.

Reply:
No worry! Thank you for the great PR! I'll release the new version tomorrow!